integration-tests/paho/src/main/java/org/apache/camel/quarkus/component/paho/PahoResource.java

```diff
@@ -92,7 +116,41 @@ public String mqttExceptionDuringReconnectShouldSucceed() {
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     public String readThenWriteWithFilePersistenceShouldSucceed(@QueryParam("message") String message) throws Exception {
-        producerTemplate.requestBody("paho:withFilePersistence?retained=true&persistence=FILE", message);
-        return consumerTemplate.receiveBody("paho:withFilePersistence?persistence=FILE", 5000, String.class);
+        producerTemplate.requestBody("paho:withFilePersistence?retained=true&persistence=FILE&brokerUrl=" + brokerUrl("tcp"),
+                message);
+        return consumerTemplate.receiveBody("paho:withFilePersistence?persistence=FILE&brokerUrl=" + brokerUrl("tcp"), 5000,
+                String.class);
+    }
+
+    private String brokerUrl(String protocol) {
+        return ConfigProvider.getConfig().getValue("paho.broker." + protocol + ".url", String.class);
+    }
+
+    private void setKeyStore(String keystore, String password) {
+        InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(keystore);
+
+        try {
+            Files.copy(in, Paths.get(keystore), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Woudn't this copy the file to the given module's root directory, where somebody could inadvertently add it to git? Maybe under tmp or target could be better?

Reply:
yeah, I agree and prefer to create a tmp file. Thanks @ppalaga !

Reply:
I make some similar changes to paho-mqtt5 ssl tests according to @ppalaga comments. Please review https://github.com/apache/camel-quarkus/pull/3731